opentelemetry-sdk/src/opentelemetry/sdk/metrics/view.py

```diff
@@ -0,0 +1,107 @@
+# Copyright The OpenTelemetry Authors
```

Comment:
How would users define the aggregator types to be used in Views? They exist currently in the SDK. Since we expect users to create custom aggregators of their own, should we move aggregators into the API package?

Also, I believe the ViewManager should stay in the SDK. It seems like implementation specific way of handling the Views, and we also don't want to introduce the whole "loading your own ViewManager" correct?

Reply:
From discussions in the metrics SIG, I believe we are creating a "SDK-API" for now, so we don't expect users to define their own views from the API.

Reply:
Writing my understanding so you can correct me if wrong:

OT API is for library authors to take dependency on. They use Metric API to emit metrics using instruments. They don't have any say in how the metrics are going to be aggregated (other than knowing the default aggregation). Views are not required to emit metrics, and hence not required to be part of API.

Views are supposed to be defined by the application owner  - and application must have SDK installed to have metrics work. And they can define Views, as Views are part of SDK.

Reply:
>They use Metric API to emit metrics using instruments.
The Metric API is used to _record_ and _create_ metrics.

>They don't have any say in how the metrics are going to be aggregated (other than knowing the default aggregation).
Since aggregation is an SDK concept, users that take dependency only on the API have no idea how to do this regardless of whether or not Views is part of the API.

>Views are not required to emit metrics, and hence not required to be part of API.
Although the first part is true, it's not really the reason why the second part is true.We don't require Views to be part of the API because we don't expect users to define their own Views (like how Meter and Tracer are used).

>Views are supposed to be defined by the application owner - and application must have SDK installed to have metrics work. And they can define Views, as Views are part of SDK.
Yes this is true., although the whole API SDK seperation doesn't really make too much sense in terms of metrics anyways. What use is just have a dependency on the API?

Reply:
My view of this separation is that there are different concerns. The "operator" is the person who knows which aggregations are useful for monitoring a system, who is different from the "developer" who writes the code. There are two separate APIs here, one for operators (Views) and one for developers (Metrics).